### PR TITLE
integr8ly: protect only master & release branches

### PR DESCRIPTION
**Pull request by psturc**
### Summary

This PR should stop OpenShift CI from re-enabling branch protection for all branches except `master` and `release-*`

### Motivation
https://issues.redhat.com/browse/MGDAPI-374

### Short summary
One of the ci-cd jobs in our internal jenkins is going to be used for rebasing commits from master branch to a specific branch ([3scale-next-0.6.1](https://github.com/integr8ly/integreatly-operator/branches)) (on a nightly basis). If this branch is protected by openshift-ci, the jenkins job won't have permissions to force-push to it.

### CODE REVIEW

File: core-services/prow/02_config/_config.yaml [hunk 1]
```diff
@@ -58,6 +58,13 @@ branch-protection:
       repos:
         toolchain-operator:
           protect: false
+    integr8ly:
+      repos:
+        integreatly-operator:
+          exclude:
+          - master
+          - release-*
+          protect: false
```
Comment (mikenairn) [inline core-services/prow/02_config/_config.yaml:67]: This will turn off branch protection for everything but master and our release branches right?

Comment (psturc) [reply core-services/prow/02_config/_config.yaml:67]: Correct. That's based on the config for [operator-sdk repo](https://github.com/openshift/release/blob/a7e608c7393e4d907e5603d74f12e1515b457eff/core-services/prow/02_config/_config.yaml#L340-L344).

[This is the result](https://github.com/operator-framework/operator-sdk/branches)

**Comment by mikenairn** [comment]
/lgtm

Will need approval from someone in the DPTP team

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**